src/isomorphic/classic/types/ReactPropTypes.js

```diff
@@ -114,6 +115,11 @@ function createChainableTypeChecker(validate) {
   ) {
     componentName = componentName || ANONYMOUS;
     propFullName = propFullName || propName;
+    if (__DEV__) {
```

Comment:
Is this check necessary? I'm pretty sure that prop types are only validated in `__DEV__` anyways so this would already be exclusively in the `__DEV__` code path.